### DISCUSSION TITLE
ESP32 S3 and ES8388

### DISCUSSION THREAD

**Discussion by grender007** [pschatzmann/arduino-audiokit#114] [Q&A] [answered]
Hi!
I've been trying to utilize PCB Artist ES8388 as an external ADC for my Feather ESP32 S3, but I had some issues configuring the pins. I already read the datasheet of ES8388, and I feel pretty comfortable with it. However, I'm a bit confused about how to implement AudioKit library to my code. I played with the input.ino example but I didn't get any results. Also I read another post with a similar problem (ESP32 S3 and this codec), but it wasn't helpful. I already set board 10 (generic ES8388), but I'am stuck. Any suggestions please?

**Comment by pschatzmann** [comment, marked as answer]
Did you read the info in the Wiki for the existing pins and the definition of new boards.

I was never quite happy with this library and therefore I started yesterday with a [new more flexible approach](https://github.com/pschatzmann/arduino-audio-driver). Would you be willing to help with the tests ?

**Reply by grender007** [reply]
Yes, I read the info in the Wiki, but I wasn't sure to manually override the pins (I'm using Arduino IDE and I'm new at this). However, I would gladly use this new library for my project and upload my code. I already worked with AudioTools and it was really helpful!

**Comment by pschatzmann** [comment]
I just committed some corrections to my [new audio-driver](https://github.com/pschatzmann/arduino-audio-driver) project and it seems that it's starting to work...

**Reply by pschatzmann** [reply]
I have also added some examples to the audio-driver project that should help you to get started. It is quite flexible so you can (optionally) define the pins in the board.

**Reply by grender007** [reply]
Thank you very much! I will update my results soon!
